src/types/k8s.go

```diff
@@ -91,6 +109,43 @@ type GitServerInfo struct {
 	InternalServer bool   `json:"internalServer" jsonschema:"description=Indicates if we are using a git server that Zarf is directly managing"`
 }
 
+// FillInEmptyValues sets every necessary value that's currently empty to a reasonable default
+func (gs GitServerInfo) FillInEmptyValues() (GitServerInfo, error) {
```

Comment:
make these pointer receivers and return just an `err` for these `FillInEmptyValues` operations imo

```go
func (gs *GitServerInfo) FillInEmptyValues() error {
```

Reply:
might be worth researching struct default value population patterns in Go

Reply:
Yeah I was 50/50 on doing that. I can see how the pointer receiver may read more naturally, though I thought since we generally use this on initopts objects that we are going to reassign anyway, the usage of value receivers was a bit nicer

Reply:
Updated to use the pointer receiver method.

Not able to find much for relevant patterns given that for this we're generally starting with an object with some values already filled in by the user. For what it's worth, chatGPT said that the pointer receiver way of doing things was more idiomatic for go.